### DISCUSSION TITLE
Seeking Advice on Setting Initial Conditions for Multi-Step Simulation in MOOSE

### DISCUSSION THREAD

**Discussion by BrownieLL** [idaholab/moose#27313] [Q&A General] [answered]
Hello everyone,

I'm conducting a simulation involving two steps. I'm encountering a problem of setting the initial conditions for step 2.

In step 1, I focused on variables such as A and B. In step 2, however, the variables shift to C and D, with C and D being the auxiliary variables from step 1.

I'm wondering if it's possible to use the auxiliary variables from step 1 as the initial conditions for the variables in step 2?

Thank you in advance for your time and assistance.

Best regards,
Xin

**Comment by GiudGiud** [comment, marked as answer]
hello

We typically do not do this in a single application solve, though it could be done with a user object setting the initial condition of nonlinear variables. I think the better way to do this is with two solves in a multiapp setup. 
See the documentation for the types of multiapps
https://mooseframework.inl.gov/syntax/MultiApps/index.html
and the tutorial on multiapps
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial02_multiapps/presentation/index.html#/

Guillaume

**Reply by BrownieLL** [reply]
Thank you for the advice!

**Reply by BrownieLL** [reply]
Hi Guillaume,

I encountered a puzzling issue. When using parallel computing, I found that setting the number of parallel processes to 12 takes more wall time than when it's set to 4. I want to know what could be the possible reasons behind this, or do you have any suggestions regarding setting the parallel computing?

Thank you!

Best regards,
Xin

**Reply by GiudGiud** [reply]
Hello

We recommend having at least 20k DoFs per process for the parallel communication costs in MOOSE to be outweighted by the computation costs. But some preconditioners will simply not scale to more than a few processes.
Having a solution method that solves the problem is a good first step, scaling to larger machines is a distinct piece of work

Guillaume

**Reply by BrownieLL** [reply]
Thank you very much!
